projects/wave-core-new/src/lib/datasets/dataset.service.ts

```diff
@@ -17,4 +19,12 @@ export class DataSetService {
             map((dataSetDicts) => dataSetDicts.map((dict) => DataSet.fromDict(dict))),
         );
     }
+
+    upload(form: FormData): Observable<HttpEvent<UploadResponseDict>> {
+        return this.userService.getSessionStream().pipe(mergeMap((session) => this.backend.upload(session.sessionToken, form)));
```

Comment:
You will upload again if the session changes, so you should use getSessionTokenForRequest or something that just emits once.

Reply:
d8c6ed229655aa08bd88a0cb90acbe9d60086637